sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -244,7 +244,11 @@ final class DataFrameWriter private[sql](df: DataFrame) {
       bucketSpec = getBucketSpec,
       options = extraOptions.toMap)
 
-    dataSource.write(mode, df)
+    dataSource.providingClass.newInstance() match {
+      case jdbc: execution.datasources.jdbc.DefaultSource =>
```

Comment:
I agree and admit I was being lazy in not trying to figure out how to make the current implementation return a `BaseRelation`. I'll take another look today at just turning the DefaultSource into a `CreatableRelationProvider`